content/en/blog/2021-09-29-server-side-reconciliation-is-coming.md

```diff
@@ -0,0 +1,181 @@
+---
+author: dholbach
+date: 2021-09-28 12:30:00+00:00
+title: Server-side reconciliation is coming
+description: >
+  The next Flux release will bring you a new reconciler based on Kubernetes "Server-Side Apply". It will make Flux more performant, observable, less error-prone and provide a generally more delightful experience. This post informs you of the changes you need to take to be able to upgrade.
+url: /blog/2021/09/server-side-reconciliation-is-coming/
+---
+
+
+**tl;dr**: Server-side reconciliation will make Flux more performant,
+improve overall observability and going forward will allow us to add new
+capabilities, like being able to preview local changes to manifests
+without pushing to upstream.
+
+⚠ **Changes required**: We need Kubernetes v1.18.8 as a
```

Comment:
I think we need to rephrase this a little, @SomtochiAma found out that SSA works on these older versions too: `v1.16.13` and `v1.17.17` but is broken on `v1.19.0`, works on `>1.19.1`.

Reply:
@SomtochiAma Did you find a k/k upstream issue for this?

Reply:
Nevermind, @stefanprodan pointed me to to https://github.com/fluxcd/kustomize-controller/pull/426#issuecomment-918384882

Thanks everyone for your attention to detail on this!

Reply:
Maybe we could list all the supported version in a table view?

Reply:
Sure... I'll review the state of things in the upstream issue.

Reply:
Fixed in
- 1.6.11: https://github.com/kubernetes/kubernetes/commit/6ca136fec3c4145947857d62a17ad715096b20e7#diff-9d54d83dc88cdae401373933ea7704c83d8ccacd8db6d53559094c4e12c46b94R258 
- 1.17.7: https://github.com/kubernetes/kubernetes/commit/03204511d2c4d3848bff0d57abec88ff655b0aac#diff-5c5c97640301c6971540da11362d147741b3ddf85bab9012a9bb7b0cb88cf60bR231
- 1.18.4: https://github.com/kubernetes/kubernetes/commit/2ff66c4524da7ac61bc9938e5014534c67ebda96#diff-8474cd59fc972c229839e00293eed54fd5729dfd9738adb48c577b0f0be4f1a6R194
- 1.19 release

Reply:
Not 100% sure if all the other required functionality is already present in these release. This is just about backports of the fix in k/k#91748.

Reply:
@SomtochiAma Stefan suggested we 1) add a small table to the blog post which indicates which versions are safe to use and 2) we add this as a version check in the flux pre-flight checks.

Shall we just say `>=1.6.11, >=1.17.7, >=1.18.4, >=1.19` since there's no real discussion upstream in k/k on when the regression was introduced?

Reply:
Right now the minimum is defined here, right?

https://github.com/fluxcd/flux2/blob/main/cmd/flux/testdata/check/check_pre.golden#L3